Editable Session Rates

wisemandenny opened the pull request:
This PR adds the ability to edit all fields in the sessions list view.
#376

wisemandenny committed:
session rates are now editable

wisemandenny committed:
session rates are now editable

wisemandenny committed:
Delete instructors (#400)

* feature, delete instructor adds delete icon col

* added dialog toggled by delete buttons

* delete button deletes instructors

siefkenj commented on frontend/src/components/sessions.js:
```diff
@@ -45,3 +91,10 @@ SessionsList.propTypes = {
+
+export const ConnectedSessionsList = connect(
+    (state) => ({
+        sessions: sessionsSelector(state),
```
Let's switch this to the `useReducer` and `useDispatch` style.

wisemandenny replied:
Do you mean `useSelector` from redux hooks or `useReducer` from React hooks?

siefkenj replied:
From `Redux`

siefkenj committed:
Add instructor routes (#404)

Add /api/v1/instructor routes and basic tests. Instructor routes first verify that the active user is an instructor and then only returns data relevant to that instructor.

siefkenj committed:
Minor fixes

siefkenj committed:
Merge pull request #405 from siefkenj/master

Minor fixes

wisemandenny committed:
session names and rates are editable with redux hooks

wisemandenny committed:
Merge branch 'editable-sessions-rates' of https://github.com/wisemandenny/tapp into editable-sessions-rates

wisemandenny commented:
~@siefkenj On this branch try editing and saving either rate on a session with the console open, you will see a "can't perform a state update on an unmounted component" error. I am not sure how to fix this, and also the same error doesn't appear when I edit the name field. Could you take a look at it and let me know if you know how to fix it? I am new to redux hooks.~

Just saw that you used `useSelector` in `editable-instructors-list.tsx`. I figured this issue out.

wisemandenny committed:
fixed error in useSelector implementation

wisemandenny committed:
can now edit dates using date picker
